crates/aggchain-proof-types/src/imported_bridge_exit.rs

```diff
@@ -1,14 +1,37 @@
+use std::cmp::Ordering;
+
 use agglayer_interop::types::GlobalIndex;
 use serde::{Deserialize, Serialize};
 
 // TODO: move this to interop repository
 // https://github.com/agglayer/provers/issues/141
-#[derive(Deserialize, Serialize, Clone, Copy, Debug)]
+#[derive(Deserialize, Serialize, Clone, Copy, Debug, Eq, PartialEq)]
 pub struct BridgeExitHash(pub agglayer_interop::types::Digest);
 
-#[derive(Deserialize, Serialize, Clone, Debug)]
+#[derive(Deserialize, Serialize, Clone, Debug, Eq, PartialEq)]
 pub struct ImportedBridgeExitWithBlockNumber {
     pub block_number: u64,
     pub bridge_exit_hash: BridgeExitHash,
     pub global_index: GlobalIndex,
 }
+
+impl PartialOrd for ImportedBridgeExitWithBlockNumber {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for ImportedBridgeExitWithBlockNumber {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.block_number.cmp(&other.block_number).then_with(|| {
+            let ordering = self.global_index.cmp(&other.global_index);
+            // Assert that if block_number and global_index are equal,
+            // then bridge_exit_hash should also be equal to maintain Ord guarantees.
+            assert!(
+                ordering != Ordering::Equal || self.bridge_exit_hash == other.bridge_exit_hash,
+                "Items with same block_number and global_index must have the same bridge_exit_hash"
+            );
```

Comment:
is this assumptions always true?¿, why we order them by global index, and not by transaction index?
Also coudl happen in case there's an unset claim, that same global index is claimed but have different or the same hash

Reply:
also just check it
all the rest we sort by block_index,
crates/aggchain-proof-types/src/imported_bridge_exit.rs it's the only one htat we use global_index and we shouldn't

Finally...
if a single transaction make 2 updates.... that would be hard to check... i guess the only way would be to retrieve the tx hash and check the order of the events... but better to reorder in a "random" way i would respect the order were this is sent... i guess is better than nothing

Reply:
@atanmarko